src/nvim/ex_docmd.c

```diff
@@ -6788,27 +6800,59 @@ void free_cd_dir(void)
 
 #endif
 
-/*
- * Deal with the side effects of changing the current directory.
- * When "local" is TRUE then this was after an ":lcd" command.
- */
-void post_chdir(int local)
-{
+/// Deal with the side effects of changing the current directory.
+///
+/// The value of `scope` reflects the name under which the command was invoked:
+///   - `SCOPE_EDITOR` for `:cd` and `chdir`
+///   - `SCOPE_TABPAGE` for `:tcd` and `tchdir`
+///   - `SCOPE_WINDOW` for `:lcd` and `lchdir`
+///
+/// @param scope  Scope of the function call (editor, tab or window).
+///
+/// @pre  The value of `scope` must be less than the number of scopes
+///       (`NUMBER_OF_SCOPES`) and higher than zero.
```

Comment:
@HiPhish 

1. Do not use `enum {name}`. Use `typedef enum { … } CdScope;`.
2. `typedef` must occur before including generated declarations.
3. For non-static functions typedef must happen in the header (BTW why `post_chdir` is not static?), again before including generated declarations.